src/postprocess.ts

```diff
@@ -52,70 +41,96 @@ function createBabelTransformer(
   options: BabelTransformOptions,
 ): PostProcessHook {
   importBabelDeps();
-  options = {
-    ...options,
-    plugins: options.plugins || [],
-    presets: (options.presets || []).concat([jestPreset]),
-  };
-  delete options.cacheDirectory;
-  delete options.filename;
+  const presets = options.presets.slice();
+  const plugins = options.plugins.slice();
+
+  // adds babel-preset-jest if not present
+  if (!hasBabelAddon(presets, jestPreset)) {
+    presets.push(jestPreset);
+  }
 
-  return (
-    codeSourcemapPair: CodeSourceMapPair,
+  // we need to know if there is istanbul plugin in the list so that we do not add it
+  // in the case the user already has it configured
+  const hasIstanbul = hasBabelAddon(plugins, istanbulPlugin);
```

Comment:
well, I don't think many users have that of course, but I am doing myself in my `.babelrc.js` a `require` of some plugins and/or presets so that I can give myself some more options, and also use some old presets not compatible with new babel

Reply:
Alright fair enough